### PR TITLE
Add `time_budget_s` parameter

### PR DESCRIPTION
**Pull request by krfricke**
This passes the `time_budget_s` parameter to `tune.run()`.

This also refactors parts of the base search `_format_results()` method to deal with missing values from trials (that e.g. have not been received due to early stopping/timeout).

### CODE REVIEW

File: tune_sklearn/tune_search.py [hunk 1]
```diff
@@ -258,6 +258,9 @@ class TuneSearchCV(TuneBaseSearchCV):
             determined by 'Pipeline.warm_start' or 'Pipeline.partial_fit'
             capabilities, which are by default not supported by standard
             SKlearn. Defaults to True.
+        time_budget_s (int|float|datetime.timedelta): Global time budget in
+            seconds after which all trials are stopped. Can also be a
+            ``datetime.timedelta`` object.
```
Comment (Yard1) [inline tune_sklearn/tune_search.py:263]: We may want to expand this docstring to explain when exactly they are stopped - at the end of the trial, at the start, in the middle, etc.

File: tune_sklearn/tune_search.py [hunk 2]
```diff
@@ -591,25 +596,25 @@ def _tune_run(self, config, resources_per_trial):
...
-                reuse_actors=True,
-                verbose=self.verbose,
-                stop=stop_condition,
-                num_samples=self.num_samples,
-                config=config,
-                fail_fast=True,
-                resources_per_trial=resources_per_trial,
-                local_dir=os.path.expanduser(self.local_dir),
-                loggers=self.loggers)
+        run_args = dict(
```
Comment (Yard1) [inline tune_sklearn/tune_search.py:599]: Great idea, been wanting to put the args in a dict myself at some point :+1:

**Review by Yard1 — CHANGES_REQUESTED** [review]
Looks good! Can we also get a test for random search? Just in case

**Comment by krfricke** [comment]
Added!

**Review by Yard1 — APPROVED** [review]

**Pull request merged**